docs/tutorials/data_management.md

```diff
@@ -0,0 +1,243 @@
+
+This will generate a `my-data` file containing the serialized data.
+
+# 3. Using the generated data
+
+Once we have generated some data, it needs to be loaded as a data provider. The blob format we chose can be loaded by `BlobDataProvider` from the `icu_provider_blob` crate.
+
+This provider performs deserialization, so it's a `BufferProvider`. This means that the feature `"serde"` needs to be enabled on `icu`.
+
+Let's update our `Cargo.toml`:
+
+```console
+$ cargo add icu --features serde
+$ cargo add icu_provider_blob
+```
+
+We can then use the provider in our code:
+
+```rust
+use icu::locid::{locale, Locale};
+use icu::calendar::DateTime;
+use icu::datetime::{DateTimeFormatter, options::length};
+use icu_provider_blob::BlobDataProvider;
+
+const LOCALE: Locale = locale!("ja");
+
+fn main() {
+    let blob = std::fs::read("my-data").expect("Failed to read file");
+    let buffer_provider = 
+        BlobDataProvider::try_new_from_blob(blob.into_boxed_slice())
+            .expect("Failed to initialize Data Provider.");
+
+    let options = length::Bag::from_date_time_style(length::Date::Long, length::Time::Medium);
+
+    let dtf = DateTimeFormatter::try_new_with_buffer_provider(&buffer_provider, &LOCALE.into(), options.into())
+        .expect("Failed to initialize DateTimeFormatter");
+
+    let date = DateTime::try_new_gregorian_datetime(2020, 10, 14, 13, 21, 28)
+        .expect("Failed to create a datetime.");
+    let date = date.to_any();
+
+    let formatted_date = dtf.format(&date).expect("Formatting should succeed");
+
+    println!("📅: {}", formatted_date);
+}
+```
+
+*Notice:* We used the `with_buffer_provider` constructor (`DateTimeFormatter::try_new_with_buffer_provider`) now, instead of the `with_any_provider` constructor that we used with test data, as we now have a `BufferProvider`. 
+
+# 4. Data slicing
+
+You might have noticed that the blob we generated is a hefty 13MB. This is no surprise, as we included `--all-keys` `--all-locales`. However, our binary only uses date formatting data in Japanese. There's room for optimization:
+
+```console
+$ icu4x-datagen --cldr-tag latest --icuexport-tag latest --out my-data --format blob --keys-for-bin target/debug/myapp --locales ja
+```
+
+The `--keys-for-bin` argument tells `icu4x-datagen` to analyze the binary and only include keys that are used by its code. In addition, we know that we only need data for the Japanese locale. This significantly reduces the blob's file size, to 54KB, and our program still works. Quite the improvement!
+
+But there is more to optimize. You might have noticed this in the output of the `icu4x-datagen` invocation, which lists 21 keys, including clearly irrelevant ones like `datetime/ethopic/datesymbols@1`. Remember how we had to convert our `DateTime<Gregorian>` into a `DateTime<AnyCalendar>` in order to use the `DateTimeFormatter`? Turns out, using `DateTimeFormatter` pulls in data for all calendar types that it supports. 
```

Comment:
```suggestion
But there is more to optimize. You might have noticed this in the output of the `icu4x-datagen` invocation, which lists 21 keys, including clearly irrelevant ones like `datetime/ethopic/datesymbols@1`. Remember how we had to convert our `DateTime<Gregorian>` into a `DateTime<AnyCalendar>` in order to use the `DateTimeFormatter`? Turns out, using `DateTimeFormatter` with `--keys-for-bin` pulls in data for all calendar types that it supports. 
```

DTF itself doesn't pull in extra data

Reply:
IMO we should also mention that the keys can be manually specified. We don't have much docs for how to pick keys, but it's worth mentioning that it's possible and saying better docs on that will come later.

Reply:
I mention it off-hand in the next paragraph. I'll add a paragraph to say look at `--help`

Reply:
Actually I didn't include it on purpose. I was going to use `--keys` for these examples, but it's a major pain to figure out which keys a constructor uses.

Reply:
Yeah I think we should _mention_ it and also say that right now it's hard to figure out which key a constructor uses beyond just using keyextract, and say we plan to improve this.

I've done this kind of thing "you can do X! it's hard right now!" in the C++/JS docs too, it's good to give people something to explore instead of not telling them it's possible.